src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5293,7 +5293,13 @@ and TcStmtThatCantBeCtorBody (cenv: cenv) env tpenv synExpr =
 and TcStmt (cenv: cenv) env tpenv synExpr =
     let g = cenv.g
     let expr, ty, tpenv = TcExprOfUnknownType cenv env tpenv synExpr
-    let m = synExpr.Range
+
+    let rec getInnerRange synExpr =
+        match synExpr with
+        | SynExpr.Sequential(expr2 = expr2)  -> getInnerRange expr2
+        | synExpr -> synExpr.Range
+    
+    let m = getInnerRange synExpr
```

Comment:
Could this be done here (inside `UnifyUnitType`)

https://github.com/dotnet/fsharp/blob/9ba6d0c6056471e1d74ea46b87cca3b06e2fb105/src/Compiler/Checking/CheckExpressions.fs#L743-L760

only once we know we want the more targeted range?

Reply:
Yeah I think we can do that. I might impact more cases as `UnifyUnitType` is called from different places. Let see what the CI says